### PR TITLE
C++: Handle nonreturning functions in IR generation

### PR DESCRIPTION
**Pull request by rdmarsh2**

### CODE REVIEW

File: cpp/ql/lib/change-notes/2023-05-02-ir-noreturn-calls.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: majorAnalysis
+---
+* In the intermediate representation, nonreturning calls now have the `Unreached` instruction for their containing function as their control flow successor. This should remove false positives involving such calls.
```
[the remark this answers is not in the document]
Comment (jketema) [reply cpp/ql/lib/change-notes/2023-05-02-ir-noreturn-calls.md:4]: I agree with Mathias here.

**Review by jketema — COMMENTED** [review]
Code looks good to me. DCA analysis times look good. The alert changes for git make sense to me. I've a hard time grasping the OpenTTD alert change from just the DCA output.

**Comment by jketema** [comment]
> Looking at this locally now so I can see the dataflow path.

This also makes sense to me. It's not completely clear to me why this path was missing before, but I don't want to dwell on that for too long unless you tell me otherwise. I do see some functions calls in the function with the sink that would now cause an unreachable instruction to be generated.

**Comment by MathiasVP** [comment]
Yeah, I don't think we need to dwell too much into it either.

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**